ADA/databricks_rstudio_sql_warehouse.qmd

```diff
@@ -1,105 +1,191 @@
+
+---
+
+### Setting up the ODBC driver
+
+An ODBC driver is required for the `ODBC` package in R to work - you must install it before attempting to use the package to connect to your data. 
+
+---
+
+#### Install the Simba Spark ODBC driver from the Software Centre
+
+---
+
+-   Open the Software Centre via the start menu
+
+-   In the 'Applications' tab, click `Simba Spark ODBC Driver 64-bit`
+
+::: {align="center"}
+![](../images/databricks-software-centre.png)
+:::
+
+-   Click install
 
 ------------------------------------------------------------------------
 
-# Downloading an ODBC driver
+### Establishing an RStudio connection using environment variables
 
 ------------------------------------------------------------------------
 
-1.  Install the 'Simba Spark ODBC' driver from the software centre.
+The `ODBC` package in RStudio allows you to connect to Databricks by creating and modifying three environment variables in your .Renviron file.
 
-    i)  Open the Software Centre via the start menu.
+To set the environment variables, call `usethis::edit_r_environ()`. You will then need to enter the following information:
 
-    ii) In the 'Applications' tab, click 'Simba Spark ODBC Driver 64-bit'. ![](../images/databricks-software-centre.png)
+```         
+DATABRICKS_HOST=<databricks-host>
+DATABRICKS_SQL_PATH=<sql-warehouse-path>
+DATABRICKS_TOKEN=<personal-access-token> 
+```
 
-    iii) Click install.
+Once you have entered the details, save and close your .Renviron file and restart R (Session > Restart R).
 
-2.  Get connection details for the SQL Warehouse from Databricks. To set up the connection you will need a few details from a SQL Warehouse within DataBricks.
+::: callout-note
+Everyone in your team that wishes to connect to the SQL Warehouse in Databricks and run your code must set up their .Renviron file individually, otherwise their connection will fail.
+:::
 
-    i)  Login to [Databricks](https://adb-6882499576863257.17.azuredatabricks.net/?o=6882499576863257)
+The sections below describe where to find the information needed for each of the four environment variables.
 
-    ii) Click on the 'SQL Warehouses' tab in the sidebar. ![](../images/databricks-SQL-warehouses.png)
+------------------------------------------------------------------------
 
-    iii) Click on the name of the warehouse you want to connect to, and click the 'connection details' tab.
+#### Databricks host
 
-    iv) Make a note of the 'Server hostname', 'Port', and 'HTTP Path'.
+------------------------------------------------------------------------
+
+The Databricks host is the instance of Databricks that you want to connect to. It's the URL that you see in your browser bar when you're on the Databricks site and should end in "azuredatabricks.net" (ignore anything after this section of the URL).
+
+------------------------------------------------------------------------
+
+#### Databricks SQL Warehouse Path
```

Comment:
I think something near the beginning needs editing, as a reader, the intro suggests to me I can't use a SQL warehouse with R code but here we are saying that's what to use with R studio?

Reply:
Ok so you can use a SQL warehouse if you're running SQL scripts from inside RStudio - will make this clearer